lib/puppet/type/file/source.rb

```diff
@@ -350,7 +350,12 @@ def chunk_file_from_source
         overwritten.
     EOT
 
-    defaultto :ignore
     newvalues(:use, :use_when_creating, :ignore)
+     munge do |value|
+      if @resource.file && @resource.line && !value.nil?
+        Puppet.puppet_deprecation_warning("The `source_permissions` parameter is deprecated. Explicitly set `owner`, `group`, and `mode`.", file: @resource.file, line: @resource.line)
+      end
+      value.nil? ? :ignore : value.to_sym
+     end
```

Comment:
I think we should make an exception for `ignore`, since it's effectively `nil`, and it looks like a non-trivial amount of modules on the forge set it to `ignore`, especially for windows modules. /cc @Iristyle @glennsarti 

```
$ ag --ignore spec --ignore stdlib --ignore vendor --ignore gem 'source_permissions\s*=>' *
bodgit-openldap-1.1.1/manifests/server/install.pp
72:    source_permissions => use_when_creating,

echomike80-tomcats-1.3.3/manifests/windows/install.pp
197:    source_permissions => ignore,
204:    source_permissions => ignore,
211:    source_permissions => ignore,
218:    source_permissions => ignore,
225:    source_permissions => ignore,
241:    source_permissions => ignore,

echomike80-tomcats-1.3.3/manifests/windows/service/tanuki.pp
88:    source_permissions => ignore,
104:    source_permissions => ignore,

echomike80-tomcats-1.3.3/manifests/windows/service/yajsw.pp
63:    source_permissions => ignore,
76:    source_permissions => ignore,

ferventcoder-powershell_net4-0.1.0/manifests/init.pp
29:    source_permissions => ignore,

gabe-choco_app-0.3.0/manifests/init.pp
11:  File { source_permissions => ignore, }

gildas-sqlserver-0.2.3/manifests/init.pp
61:  File { source_permissions => ignore }

grundic-teamcity-1.1.1/manifests/agent/install.pp
37:      source_permissions => ignore,

jriviere-windows_services-0.0.4/manifests/credentials.pp
43:    source_permissions => ignore,

jriviere-windows_sharepoint-0.0.7/manifests/prepsp.pp
31:    source_permissions => ignore,
35:    source_permissions => ignore,
39:    source_permissions => ignore,
43:    source_permissions => ignore,
48:    source_permissions => ignore,
52:    source_permissions => ignore,

jriviere-windows_sharepoint-0.0.7/manifests/warmup.pp
19:      source_permissions => ignore,

jriviere-windows_sql-0.0.9/manifests/maintenance/backup.pp
85:      source_permissions => ignore,
98:      source_permissions => ignore,

jriviere-windows_sql-0.0.9/manifests/maintenance/deletebackuphistory.pp
70:      source_permissions => ignore,

jriviere-windows_sql-0.0.9/manifests/maintenance/indexoptimize.pp
80:      source_permissions => ignore,
93:      source_permissions => ignore,

jriviere-windows_sql-0.0.9/manifests/maintenance/integritycheck.pp
79:      source_permissions => ignore,
93:      source_permissions => ignore,

maestrodev-maestro-2.0.7/manifests/agent/user.pp
33:        source_permissions => ignore,

mdelaney-cygwin-0.0.6/manifests/install.pp
60:      source_permissions => ignore,

noma4i-windows_cis-0.1.3/manifests/init.pp
6:    source_permissions => ignore,

noma4i-windows_tableau-0.2.0/manifests/init.pp
10:      File { source_permissions => ignore }
14:          source_permissions => ignore,
41:        source_permissions => ignore

noma4i-windows_updates-0.2.1/manifests/init.pp
5:    source_permissions => ignore,
11:    source_permissions => ignore

noma4i-windows_xmltask-0.0.3/manifests/init.pp
13:      source_permissions => 'ignore',

PierrickL-cicserver-1.2.6/manifests/install.pp
365:          source_permissions => ignore,
375:          source_permissions => ignore,

PierrickL-glance-0.1.0/manifests/init.pp
65:    File { source_permissions => ignore } # Required for windows

PierrickL-vidyo-0.1.6/manifests/init.pp
133:    File { source_permissions => ignore } # Required for windows

pltraining-classroom-1.5.9/files/classroom.pp
12:    source_permissions => ignore,

puppetlabs-puppet_agent-1.0.0/manifests/prepare.pp
10:      source_permissions => ignore,
15:      source_permissions => use,

souldo-pe_winagent-2.0.2/manifests/powershell_host.pp
6:  File { source_permissions => ignore }

tobiashochguertel-ps1template-0.2.3/Dokumentation/example - deploy.pp
4:    File { source_permissions => ignore }

tobiashochguertel-ps1template-0.2.3/Dokumentation/example - undeploy.pp
4:    File { source_permissions => ignore }

wazuh-ossec-2.0.5/manifests/client.pp
```

Reply:
Yeah I'd agree with that. To summarise;

Setting ` source_permissions => ignore` is kind of the same as not setting source_permissions at all so it shouldn't require a deprecation message.

Although...Is the intent to remove this property in Puppet 6/7?  If so, then a deprecation warning may be required?

Reply:
We can't eliminate it yet because it's still used internally to pluginsyncing external executable facts. So I think only issuing a deprecation warning for `use` and `use_when_creating` makes sense.